addOns/insights/src/main/java/org/zaproxy/addon/insights/StatsMonitor.java

```diff
@@ -0,0 +1,246 @@
+        if (key.startsWith(STATS_CODE_PREFIX)) {
+            return Constant.messages.getString(
+                    ExtensionInsights.PREFIX + ".insight.stats.code",
+                    key.substring(STATS_CODE_PREFIX.length()));
+        }
+        return key;
+    }
+
+    private long unbox(Long l) {
+        return l == null ? 0 : l;
+    }
+
+    private void recordNonZeroInsight(Insight.Level level, String key) {
+        long val = unbox(stats.getStat(key));
+        if (val > 0) {
+            recordInsight(Insight.Level.Low, key, val);
+        }
+    }
+
+    private void recordInsight(Insight.Level level, String key, long stat) {
+        this.recordInsight(level, "", key, stat);
+    }
+
+    private void recordInsight(Insight.Level level, String site, String key, long stat) {
+        ext.recordInsight(new Insight(level, site, key, getStatsDescription(key), stat));
+    }
+
+    public void processStats() {
+        // Count of responses by site and status code
+        Map<String, Map<String, Long>> siteCodeStats = stats.getAllSiteStats(STATS_CODE_PREFIX);
+        for (Entry<String, Map<String, Long>> site2stats : siteCodeStats.entrySet()) {
+            String site = site2stats.getKey();
+            // Record all stats by
+            long total = 0;
+            long tot400 = 0;
+            long tot500 = 0;
+
+            for (Entry<String, Long> k2stat : site2stats.getValue().entrySet()) {
+                recordInsight(Insight.Level.Info, site, k2stat.getKey(), k2stat.getValue());
+                total += k2stat.getValue();
+                if (k2stat.getKey().startsWith(STATS_CODE_PREFIX + "4")) {
+                    tot400 += k2stat.getValue();
+                } else if (k2stat.getKey().startsWith(STATS_CODE_PREFIX + "5")) {
+                    tot500 += k2stat.getValue();
+                }
+            }
+
+            if (tot400 > total / 20) {
+                recordInsight(
+                        Insight.Level.Low, site, STATS_CODE_PREFIX + "4xx", tot400 * 100 / total);
+            }
+            if (tot500 > total / 20) {
+                recordInsight(
+                        Insight.Level.Low, site, STATS_CODE_PREFIX + "5xx", tot500 * 100 / total);
+            }
+        }
+
+        // Network problems
+        long netGood = unbox(stats.getStat(STATS_NETWORK_SUCCESS));
+        long netBad = unbox(stats.getStat(STATS_NETWORK_FAILURE));
+        long netTotal = netGood + netBad;
+        if (netTotal > MIN_NUMBER_OF_REQS) {
+            if (netBad > netTotal / 2) {
+                // TODO need to remove the medium one, if present!
+                recordInsight(
+                        Insight.Level.High, "stats.network.failure.high", netBad * 100 / netTotal);
+            } else if (netBad > netTotal / 20) {
+                recordInsight(
+                        Insight.Level.Medium,
+                        "stats.network.failure.medium",
+                        netBad * 100 / netTotal);
```

Comment:
Seems like this math is done either way, why not move it outside of the conditions and give it a meaningful name?